### ISSUE TITLE
Cannot start the plugin

### ISSUE THREAD

**Issue by szafranski** [mark-bloom/OctoLight_Home-Assistant#5]
Hi, some syntax error in the plugin file.

2023-07-09 14:27:06,345 - octoprint.plugin.core - ERROR - Invalid syntax in plugin file of plugin octolightHA
Traceback (most recent call last):
  File "/home/pawel/OctoPrint/venv/lib/python3.11/site-packages/octoprint/plugin/core.py", line 800, in _parse_metadata
    return parse_plugin_metadata(self.location)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/pawel/OctoPrint/venv/lib/python3.11/site-packages/octoprint/plugin/core.py", line 78, in parse_plugin_metadata
    root = ast.parse(f.read(), filename=path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/ast.py", line 50, in parse
    return compile(source, filename, mode, flags,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/pawel/OctoPrint/venv/lib/python3.11/site-packages/octoprint_octolightHA/__init__.py", line 218
                 return dict(
                ^
IndentationError: unexpected indent
2023-07-09 14:27:06,348 - octoprint.plugin.core - WARNING - Plugin OctoLight-Home-Assistant (0.0.1) can't be compiled under Python 3.11.2 due to invalid syntax

**Comment by szafranski** [comment]
![image](https://github.com/mark-bloom/OctoLight_Home-Assistant/assets/23621888/7085f50d-1519-493f-a0c5-3fcc531ca22a)
that's what I got after correcting the indent. Maybe due to Python 3.11?

**Comment by BRFreshness** [comment]
It's not a Python 3.11 issue. I think it's a tab-versus-spaces formatting issue. People should never use tabs in Python files, IMO. They mess things up too easily.

Also, the author has not responded in three weeks.

**Comment by mark-bloom** [comment]
Please advise if the corrections to v0.0.2 have resolved your problem. Thank you for identifying the formatting problem.

All the formatting around space/tabs has been unified to spaces and the new version can be downloaded through the plugin manager (it may not update the version number immediately, but if you uninstall and reinstall, it is finding the 0.0.2).

**Comment by szafranski** [comment]
Hi, so it looks like this:
<img width="711" alt="image" src="https://github.com/mark-bloom/OctoLight_Home-Assistant/assets/23621888/3e9ae990-0a65-4c4c-9663-cbd911df7e6d">
<img width="998" alt="image" src="https://github.com/mark-bloom/OctoLight_Home-Assistant/assets/23621888/752eef9d-d33d-4ed0-b495-d9369bca7e9f">

I am happy to test more, but this seems like still not working.

**Comment by szafranski** [comment]
ok, after some digging I added "http://" and port number in HA address in plugin settings (btw - maybe it would be good to state if thats absolutely required? now I got this:
<img width="1043" alt="image" src="https://github.com/mark-bloom/OctoLight_Home-Assistant/assets/23621888/7610f298-2d51-40f9-bc58-f8f6dda34b17">

**Comment by mark-bloom** [comment]
Can you advise what entity type you are using for your light? Someone else had a similar problem using the 'switch' entity type, since the program currently supports the 'light' entity type in HA.

I can try and add some regex to add 'http' (and remove trailing slashes) at some point, although it is noted in the settings subtitles..

**Comment by mark-bloom** [comment]
v0.0.3 has added support for 'switch' entities. You can try that update and see if your issue is resolved.

**Comment by szafranski** [comment]
Ok, so:
1. I was using/trying to use plugin with light-type entity all the time, but I today changed it to switch and I forgot about it. I made test with light entity again with version 0.0.2 and it worked flawlessly - thank you!
2. Next, after seeing your following message, I updated plugin to version 0.0.3 and it now works with light as well as switch entity! Thats awesome man! Thank you! It can be now used easily with socket-type relays etc. Great job!

As for other issues:
3. Will this 0.0.3 version be available in official OctoPrint repo? For now I had to use manual installation method. Not a big deal, just asking how this repo-adding works.
4. Do you think it will be possible to add few? You could have for example 1 light and one relay connected to PSU and separate automations for those.

5. Sorry for mixing threads, but given I already started this "http" thing I will briefly explain what I mean. IMHO, if user inputs something that is surely an error (like no "http://" or so) the "save" action should be forbidden or something. I had to dig in "logs" to see what's wrong. I know - explanation is below the text but I had 2 issues with this. One - there are "examples" not "it had to be put like this". Especially, after that there is "DO NOT" which seems much more important requirement. Second, there is a statement that you have to provide "ip or hostname" not "http/https + ip or hostname + port number". It is only in examples. Maybe some greyed out example format in the filed, like you sometimes see, before you start to put anything would be beneficial? And information asking like "are you sure that data is correct", when there is no "http" and no "port number" when you saving would be a good idea? Even in logs it can be hard to find, but stated clearly would help. To be clear - I don't want to say that anything is wrong. I just thing this plugin is so awesome that more and more people will use it. I can say that I am quite familiar with networking, tinkering, raspberry, 3d printing etc. but I made that mistake, so maybe some other, less tech-savvy person could too. It would be a shame, given how cool is this plug - as I write before.    

In summary - awesome job, thank you very much for quick reaction!

**Comment by mark-bloom** [comment]
3. I don't know how long Octoprint repo takes to 'notify of plugin update', but if you uninstall and reinstall via the plugin manager, you'll see that it installs 0.0.3 (or the latest release).
4. Probably outside the scope of what I can do in a reasonable amount of time. For PSU automation I recommend using PSUControl + PSUControl-HomeAssistant add-on (I have both installed as per below)
![image](https://github.com/mark-bloom/OctoLight_Home-Assistant/assets/3166605/a086ac8f-101f-41a3-8c12-2fc55336006f)
![image](https://github.com/mark-bloom/OctoLight_Home-Assistant/assets/3166605/bf1ffaed-4dde-42f7-b5bc-3e587aea5c3b)
5. I'll look into changing the text and also adding some basic url checks. I'm unsure if I can do error checking on the input and display red text or similar (definitely possible, just don't know how to do it).

For now I'll close the issue as the switch/light issue is resolved, but I'll add your suggestions to the backlog.
